sdk/src/azure/core/az_mqtt5_rpc_server.c

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/az_mqtt5_rpc_server.h>
+#include <azure/core/az_result.h>
+#include <azure/core/internal/az_log_internal.h>
+#include <stdlib.h>
+
+#include <azure/core/_az_cfg.h>
+
+AZ_NODISCARD az_mqtt5_rpc_server_options az_mqtt5_rpc_server_options_default()
+{
+  return (az_mqtt5_rpc_server_options){
+    .subscribe_timeout_in_seconds = AZ_MQTT5_RPC_SERVER_DEFAULT_TIMEOUT_SECONDS,
+  };
+}
+
+AZ_NODISCARD az_result
+az_rpc_server_get_subscription_topic(az_mqtt5_rpc_server* client, az_span out_subscription_topic)
+{
+  (void)client;
+#ifndef AZ_NO_PRECONDITION_CHECKING
+  _az_PRECONDITION_VALID_SPAN(client->_internal.model_id, 1, false);
+  _az_PRECONDITION_VALID_SPAN(client->_internal.client_id, 1, false);
+  int32_t subscription_min_length = az_span_size(client->_internal.model_id)
+      + az_span_size(client->_internal.client_id)
+      + (az_span_size(client->_internal.command_name) > 0
+             ? az_span_size(client->_internal.command_name)
+             : 1)
+      + 23;
+  _az_PRECONDITION_VALID_SPAN(out_subscription_topic, subscription_min_length, true);
+#endif
+
+  az_span_fill(out_subscription_topic, ' ');
```

Comment:
needed?

Reply:
only needed in tests, so moved this to the buffer in the test and removed here